docs/specification.md

```diff
@@ -311,89 +432,153 @@ available state entry is `D`:
+commits that must be retained are interspersed with "bad" commits that must be
+rejected.
+
+In both cases, new RSL entries and annotations may be used to record the
+incident or to skip the invalid RSL entries corresponding to the rejected
+changes. TODO: should these new RSL entries come from authorized users for the
+affected namespace?
+
+#### M2: Create RSL entry on behalf of another user
+
+This mechanism is necessary for adoptions where a subset of developers do not
+use gittuf. When they submit changes to the main copy of the repository, they do
+not include RSL entries. Therefore, when a change is pushed to a branch by a
+non-gittuf user A, a gittuf user B can submit an RSL entry on their behalf.
+Additionally, the entry must identify the original user (TODO: cross-reference
+with signed pushes?) and include some evidence about why B thinks the change
+came from A (TODO: signed push can be evidence option?).
+
+TODO: decide if B needs to be authorized for the affected branch to sign for A.
+
+### Recovery Scenarios
+
+These scenarios are some examples where recovery is necessary.
+
+#### A change is made without an RSL entry
+
+Bob does not use gittuf and pushes to a branch. Alice notices this as her gittuf
+client detects a push to the branch without an accompanying RSL entry. She
+validates that the change came from Bob and creates an RSL entry on his behalf,
+identifying him and including information about how she verified it was him.
+Alice applies M2.
+
+#### An incorrect RSL entry is added
+
+There are several ways in which an RSL entry can be considered "incorrect". If
+an entry is malformed (structurally), Git may catch it if it's not a valid
+commit. In such instances, the push from a buggy client is rejected altogether,
+meaning other users are not exposed to the malformed commit.
+
+Invalid entries that are not rejected by Git must be caught by gittuf. Some
+examples of such invalid entries are:
+
+* RSL entry is for a non-existing Git reference
+* Commit recorded in RSL entry does not exist
+* Commit recorded in RSL entry does not match the tip of the corresponding Git
+  reference
+* RSL annotation contains references to RSL entries that do not exist or are not
+  RSL entries (i.e. the annotation points to other commits in the repository)
+
+Note that as invalid RSL entries are only created by buggy or malicious gittuf
+clients, these entries cannot be detected prior to them being pushed to the main
+repository.
+
+As correctly implemented gittuf clients verify the validity of RSL entries when
+they pull from the main repository, the user is warned if invalid entries are
+encountered. Then, the user can then use M1 to invalidate the incorrect entry.
+Other clients with the invalid entry only need to fetch the latest RSL entries
+to recover. The client that originally created the invalid entries must first
+switch to a correct implementation of gittuf and then sync the local
+repository's state with the main repository.
```

Comment:
Why does the client that made the invalid entries need to do anything different?

Reply:
Updated to only call out that the broken impl must be fixed / updated.